huaweicloud/services/eg/resource_huaweicloud_eg_connection.go

```diff
@@ -0,0 +1,386 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product EG
+// ---------------------------------------------------------------
+
+package eg
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceConnection() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceConnectionCreate,
+		UpdateContext: resourceConnectionUpdate,
+		ReadContext:   resourceConnectionRead,
+		DeleteContext: resourceConnectionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the name of the connection.`,
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the VPC to which the connection belongs.`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the subnet to which the connection belongs.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the description of the connection.`,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
can it be work if `type` is not specified?

Reply:
yes, the service will provide a default value, WEBHOOK